### DISCUSSION TITLE
IPv6 support

### DISCUSSION THREAD

**Discussion by bbergeron0** [narrowlink/narrowlink#9] [Q&A] [answered]
I wish to integrate Narrowlink into my personal private network, which currently run on IPv6. However gateway.yaml's listen_addr doesn't accept IPv6 addressed (`Error: GatewayError { error_type: "Invalid Config" }`).

I see enough value to configure IPv4+6, but I'm still curious whether support for IPv6 will be added anytime in the future.

Also, first! ;)

**Comment by SajjadPourali** [comment]
It should be supported out of the box.
Could you please wrap the IPv6 address in single or double quotes?

```yaml
listen_addr: "[::]:80"
```

Please note that the default value of `/proc/sys/net/ipv6/bindv6only` on Linux is `0`. This means that the OS will also map the IPv6 address to the IPv4 interface.

**Comment by bbergeron0** [comment, marked as answer]
Well, I forgot to enclose the address between square brackets, and I didn't specify a port. I confirm IPv6 works, and I can connect an agent to my gateway. Thanks for the hint!
